Plugins/Public/commoditylimit/Main.cpp

```diff
@@ -145,6 +166,45 @@ void ClearClientInfo(uint iClientID)
 //Functions to hook
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+void __stdcall GFGoodSell(struct SGFGoodSellInfo const& gsi, unsigned int client)
+{
+	returncode = DEFAULT_RETURNCODE;
+	const uint& baseId = Players[client].iBaseID;
+	if (!mapProducers.count(baseId))
+	{
+		return;
+	}
+
+	if (!mapProducers.at(baseId).count(gsi.iArchID))
+	{
+		return;
+	}
+	static uint voiceDealerNotInterested = CreateID("not_interested");
+	pub::Player::SendNNMessage(client, voiceDealerNotInterested);
+	sellSuppressionMap[client] = gsi.iArchID;
+	returncode = SKIPPLUGINS;
+}
+
+void __stdcall ReqRemoveItem_AFTER(unsigned short iID, int count, unsigned int client)
+{
+	returncode = DEFAULT_RETURNCODE;
+	if (!sellSuppressionMap.count(client))
+	{
+		return;
+	}
+
+	uint baseId = Players[client].iBaseID;
+	uint goodId = sellSuppressionMap.at(client);
+	float price = BaseDataList_get()->get_base_data(baseId)->market_map.find(goodId).value()->fPrice;
```

Comment:
How well does this interact with prices.cfg overrides from DSAceSrv?

Also, don't we have pre-existing buying blocking code somewhere?

Reply:
DSAceSrv updates the BaseDataList, so there is no issue here.

As for the code existing elsewhere, I found it in Alley's plugin. Moving code there.

Reply:
Note to self: Don't code past 3AM.

Alley's code was for buy prevention, not sell prevention. This code is Sell prevention, which is only present in POB plugin, and it'd be a very poor place to force this feature into.